File: vortexdbg-android/src/main/java/com/vortexdbg/linux/struct/dl_phdr_info64.java
package com.vortexdbg.linux.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class dl_phdr_info64 extends VortexdbgStructure {

    public dl_phdr_info64(Pointer p) {
        super(p);
    }

    public long dlpi_addr; // ptr
    public long dlpi_name; // ptr
    public long dlpi_phdr; // ptr
    public short dlpi_phnum;

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("dlpi_addr", "dlpi_name", "dlpi_phdr", "dlpi_phnum");
    }

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/struct/dl_phdr_info64.kt
package com.vortexdbg.linux.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class dl_phdr_info64(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var dlpi_addr: Long = 0 // ptr
    @JvmField
    var dlpi_name: Long = 0 // ptr
    @JvmField
    var dlpi_phdr: Long = 0 // ptr
    @JvmField
    var dlpi_phnum: Short = 0

    override fun getFieldOrder(): List<String> {
        return listOf("dlpi_addr", "dlpi_name", "dlpi_phdr", "dlpi_phnum")
    }

}
